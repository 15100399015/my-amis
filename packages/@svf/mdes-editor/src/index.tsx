import 'mdes';
export * from 'mdes-editor-core';
export * from './plugin';
export * from './icons/index';
export * from './renderer/index';
import './tpl/index';
