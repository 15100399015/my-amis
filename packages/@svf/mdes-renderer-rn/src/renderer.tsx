import React from 'react';
import type {ReactElement} from 'react';
// 组件 map
export class ComponentMap {
  private readonly map = new Map<string, React.ComponentType>();
  public register(name: string, element: React.ComponentType) {
    if (!this.map.has(name)) this.map.set(name.toLowerCase(), element);
  }
  public getComponent(name: string) {
    return this.map.get(name);
  }
  public hasComponent(name: string) {
    return this.map.has(name);
  }
}
export const componentMap = new ComponentMap();

export const EmptyComponent = () => null;

type RendererProps = {
  schema: any;
  /**
   * 上下文数据
   */
  contextData: any;
  /**
   * 事件
   */
  events: Record<string, Function>;
};
type RendererState = {
  element: any;
};

/**
 * 是否文本组件
 * @param item
 * @returns
 */
const isText = (item: any) => item.type === 'base-text';

/**
 * 渲染能力组件
 */
export class Rnederer extends React.PureComponent<
  RendererProps,
  RendererState
> {
  state: Readonly<RendererState> = {
    element: null
  };

  constructor(props: RendererProps) {
    super(props);
  }

  componentDidUpdate(
    prevProps: Readonly<RendererProps>,
    prevState: Readonly<RendererState>,
    snapshot?: any
  ): void {
    // schema 更新
    if (prevProps.schema !== this.props.schema) {
      this.setState({element: this.getReactElementTree()});
    }
  }

  private getReactElementTree() {
    return mapElement(this.props.schema);
  }

  render(): React.ReactNode {
    const {element} = this.state;
    return <>{element}</>;
  }
}

function mapElement(schema: any): any {
  // 如
  if (!Array.isArray(schema) || schema.length === 0) return null;
  return schema.map((item, index) => {
    const el = componentMap.getComponent(item.type);
    if (!el) return null;
    return React.createElement(
      el,
      {...item, key: index, body: null},
      isText(item) ? item.content : mapElement(item.body)
    );
  });
}
