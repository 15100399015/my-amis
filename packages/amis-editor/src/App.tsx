import * as React from 'react';
import {AlertComponent, ToastComponent, ContextMenu} from 'amis';
import AMisSchemaEditor from './Editor';

import '@fortawesome/fontawesome-free/css/all.css';
import '@fortawesome/fontawesome-free/css/v4-shims.css';

import './style.scss';

export default class App extends React.PureComponent {
  render() {
    return (
      <div className="Editor-Demo">
        <div id="headerBar" className="Editor-header">
          <div className="Editor-title">可视化</div>
        </div>
        <AMisSchemaEditor theme={'cxd'} />
        <ToastComponent theme={'cxd'} />
        <AlertComponent theme={'cxd'} />
        <ContextMenu theme={'cxd'} />
      </div>
    );
  }
}
