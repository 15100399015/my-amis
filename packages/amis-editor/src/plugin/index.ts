// react native 组件
export * from './ReactNative/Image';
export * from './ReactNative/View';
export * from './ReactNative/TouchableView';
export * from './ReactNative/Text';
export * from './ReactNative/ImageBackground';
export * from './ReactNative/LinearGradient';

export * from './Page';
