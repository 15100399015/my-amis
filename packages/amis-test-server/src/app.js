const express = require('express');
const fs = require('fs');
const cors = require('cors');
const path = require('path');
const {printQr} = require('./printQr');

const tplPath = path.join(__dirname, 'tpl.json');

const app = express();

const serverPort = 3030;

app.use(cors());
app.use(express.json());

app.post('/save', (req, res) => {
  const tpl = JSON.stringify(req.body);
  fs.writeFileSync(tplPath, tpl);
  res.status(200).send({
    status: true
  });
});

app.get('/get', (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  const data = fs.readFileSync(tplPath);
  res.status(200).send(data);
});

app.listen(serverPort, () => {
  console.log('运行成功');
});

printQr(':3030/get');
