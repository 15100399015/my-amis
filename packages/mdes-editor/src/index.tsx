import 'mdes';
export * from 'mdes-editor-core';
export * from './builder';
export * from './plugin';
export * from './icons/index';
export * from './renderer/index';
import './tpl/index';

import 'mdes-theme-editor-helper';
