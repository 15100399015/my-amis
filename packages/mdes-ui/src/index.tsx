export * from './components';
export * from './hooks';

import './locale/zh-CN';
import './themes/cxd';
import './themes/default';
import type {SchemaEditorItemPlaceholder} from './components/schema-editor/Common';
import {schemaEditorItemPlaceholder} from './components/schema-editor/Common';

export {schemaEditorItemPlaceholder, SchemaEditorItemPlaceholder};
