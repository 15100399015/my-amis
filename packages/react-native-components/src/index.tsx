export * from './components/Image';
export * from './components/ImageBackground';
export * from './components/Text';
export * from './components/View';
export * from './components/LinearGradient';
export * from './components/TouchableView';
export * from './pedestal';
