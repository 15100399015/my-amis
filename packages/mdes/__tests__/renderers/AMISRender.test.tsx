import React = require('react');
import {render} from '@testing-library/react';
import '../../src';
import {render as mdesRender} from '../../src';
import {makeEnv} from '../helper';

test('Renderer:mdesRender test', () => {
  const {container} = render(
    mdesRender(
      {
        type: 'mdes',
        value: {
          type: 'tpl',
          tpl: 'hello world'
        }
      },
      {},
      makeEnv({})
    )
  );
  expect(container).toMatchSnapshot();
});
