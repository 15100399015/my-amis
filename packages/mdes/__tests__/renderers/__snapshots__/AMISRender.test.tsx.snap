// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer:mdesRender test 1`] = `
<div>
  <span
    class="cxd-TplField"
  >
    <span>
      hello world
    </span>
  </span>
</div>
`;
